operators/src/engine/execution_context.rs

```diff
@@ -1,46 +1,161 @@
-use geoengine_datatypes::{
-    primitives::Coordinate2D,
-    raster::{GridShape2D, TilingSpecification},
-};
-
-use crate::concurrency::{ThreadPool, ThreadPoolContext};
+use crate::concurrency::ThreadPool;
+use crate::engine::{QueryRectangle, VectorResultDescriptor};
+use crate::error::Error;
+use crate::mock::MockDataSetDataSourceLoadingInfo;
+use crate::source::OgrSourceDataset;
+use crate::util::Result;
+use geoengine_datatypes::dataset::DataSetId;
+use geoengine_datatypes::raster::GridShape;
+use geoengine_datatypes::raster::TilingSpecification;
+use std::any::Any;
+use std::collections::HashMap;
+use std::fmt::Debug;
 use std::path::PathBuf;
 
-/// A context that provides certain utility access during operator execution
-/// TODO: maybe this is rather a trait
-#[derive(Debug, Clone)]
-pub struct ExecutionContext<'pool> {
-    pub raster_data_root: PathBuf,
-    pub thread_pool: ThreadPoolContext<'pool>,
-    pub tiling_specification: TilingSpecification,
+/// A context that provides certain utility access during operator initialization
+pub trait ExecutionContext:
+    Send
+    + Sync
+    + LoadingInfoProvider<MockDataSetDataSourceLoadingInfo, VectorResultDescriptor>
```

Comment:
I see it differently. Any impl of ExecutionContext needs to be LoadingInfoProvider for all Operators because it has to give LoadingInfos to all operators. That's why I demand it here. If a operator needs only one specific LoadingInfoProvider it can ask for this one, but any implementation of ExecutionContext must provide them all